pkg/kclient/ingress.go

```diff
@@ -3,15 +3,44 @@ package kclient
 import (
 	"context"
 	"fmt"
-
+	"github.com/openshift/odo/pkg/unions"
 	"github.com/pkg/errors"
 
 	extensionsv1 "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CreateIngress creates an ingress object for the given service and with the given labels
-func (c *Client) CreateIngress(ingress extensionsv1.Ingress) (*extensionsv1.Ingress, error) {
+//CreateIngress creates a specified Kubernetes Ingress as a networking v1 or extensions v1beta1 ingress depending on what
+//is supported, with preference for networking v1 ingress. The passed ingress MUST be a generated one, i.e it must
+//have been created using unions.NewKubernetesIngressFromParams
+func (c *Client) CreateIngress(ingress unions.KubernetesIngress) (*unions.KubernetesIngress, error) {
+	var err error
+	if !ingress.IsGenerated() {
+		panic("please pass a generated ingress")
```

Comment:
Instead of panicking, we can return an empty URL or an error or the caller to handle.

Reply:
But this may potentially lead to a panic for the user and that would be a very bad user experience.